src/lesson7/Shape.java

```diff
@@ -0,0 +1,28 @@
+package lesson7;
+
+public abstract class Shape {
+    protected double width;
+    protected double high;
+    public Shape() {
+
+    }
+    public Shape(double width, double high) {
+    }
+    public Shape(double high) {
+    }
+
+    public void setHigh(double high) {
+        this.high = high;
+    }
+
+    public void setWidth(double width) {
+        this.width = width;
+    }
+
+    public abstract double getPerimeter ();
+    public abstract double getArea ();
+
+    public abstract String dataAboutShape();
```

Comment:
В абстракний клас виносимо все що характерне всім фігурам, висоти і ширини не буде для кола 
```suggestion
public abstract class Shape {
    public abstract double getPerimeter ();
    public abstract double getArea ();
    public abstract String dataAboutShape();
```